### ISSUE TITLE
Specify minimum pyarrow version.

### ISSUE THREAD

**Issue by rosscoleman** [pacman82/arrow-odbc-py#42]
I tried to install in an environment that was using pyarrow==6.0.0
```
cannot import name 'RecordBatchReader' from 'pyarrow'
```

It looks like pyarrow==8.0.0 added RecordBatchReader, but pip didn't know about that minimum version. Could you add the minimum pyarrow version to pyproject.toml?

**Comment by pacman82** [comment]
Hello @rosscoleman ,

this package does not use RecordBatchReader of pyarrow at the moment. It soly relys on the C-Interface of Arrow itself. I am happy to add the minimum version, but this would be the first pyarrow version to support the arrow C-Interface, not the first to include RecordBatchReader.

Best, Markus

**Comment by rosscoleman** [comment]
I think the wheel does bring in pyarrow.
```
python -m venv arrow-odbc-test
source arrow-odbc-test/bin/activate
pip install --upgrade pip wheel
pip install arrow-odbc==1.1.1
pip install pipdeptree
```

```
$ pipdeptree
arrow-odbc==1.1.1
├── cffi [required: Any, installed: 1.15.1]
│   └── pycparser [required: Any, installed: 2.21]
└── pyarrow [required: Any, installed: 12.0.1]
    └── numpy [required: >=1.16.6, installed: 1.25.1]
pip==23.2
pipdeptree==2.10.2
setuptools==58.1.0
wheel==0.40.0
```

Then, 
```
pip install pyarrow==7.0.0
```

```
(arrow-odbc) (base) rcoleman12@imt-d-batch-ds1-1:/data/hub/rcoleman12/venvs$ python
Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:24:11) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from arrow_odbc import read_arrow_batches_from_odbc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/hub/rcoleman12/venvs/arrow-odbc/lib/python3.9/site-packages/arrow_odbc/__init__.py", line 3, in <module>
    from .writer import BatchWriter, insert_into_table, from_table_to_db
  File "/data/hub/rcoleman12/venvs/arrow-odbc/lib/python3.9/site-packages/arrow_odbc/writer.py", line 3, in <module>
    from pyarrow import RecordBatchReader
ImportError: cannot import name 'RecordBatchReader' from 'pyarrow' (/data/hub/rcoleman12/venvs/arrow-odbc/lib/python3.9/site-packages/pyarrow/__init__.py)
>>> 
```

**Comment by rosscoleman** [comment]
In short, writer.py from this package has
```
from pyarrow import RecordBatchReader
```
That makes `pyarrow>=8.0.0` the dependency.

**Comment by pacman82** [comment]
You are of course correct. I consciously avoided the dependency back then it has only been the reader but did miss this. Thanks for pointing this out.

Will add the minimum version.

Best, Markus

**Comment by rosscoleman** [comment]
Thanks!

**Comment by pacman82** [comment]
In the newly released `arrow-odbc 1.1.3` wheel. Closing this issue.

> Do you have a place for more open-ended discussion?

Enabled "discussions" in this repository. We can go talk there if you want.

> Once connector-x adds ODBC support, it looks like it will solve the same problem as this project. I'm wondering if there are possible synergies between the two projects.

Short answer: I have limited time, but if someone wants to use arrow-odbc-py to enable connector-x to work I am happy to help.
